src/services/tau/tau.cpp

```diff
@@ -33,46 +34,76 @@
 /// @file  tau.cpp
 /// @brief Caliper TAU service
 
-#include "../common/ToolWrapper.h"
-#include "caliper/common/filters/RegexFilter.h"
-
+// Caliper annotation bindings for TAU service
 
+#include "caliper/AnnotationBinding.h"
+#include "caliper/common/Attribute.h"
+#include "caliper/common/Variant.h"
 #include <map>
-#include <TAU.h>
+#define TAU_ENABLED
+#define TAU_DOT_H_LESS_HEADERS
+#include "TAU.h"
 
-namespace cali {
 
+using namespace cali;
+
+namespace
+{
+
+class TAUBinding : public cali::AnnotationBinding
+{
+    // when an attribute is created, store the value in TAU as metadata.
+    // this seems reasonable, but might be misguided.
+    static void create_attr_cb(Caliper* c, const Attribute& attr) {
+        std::string tmp("Caliper: " + attr.name());
+        Tau_metadata(tmp.c_str(), c->get(attr).value().to_string().c_str());
```

Comment:
This doesn't do what you think it does. Caliper attributes are the "keys" in our key:value pairs, like "function", "loop", etc. So this callback is just for creating a new attribute. Meanwhile `c->get(attr)` will get the current value of a key:value pair once one has been set (e.g., "main" for attribute "function"). There will never be a value while the attribute (i.e. key) is just being created.